File: ICS/CaesarCipher.java
import java.util.*;
import java.util.Arrays;
import java.util.Scanner;
import java.io.*;

class Solution{
	String caesarCipher(String text , int shiftVal){
		String result = "";
		for(int i = 0;i < text.length();i++){
			if(Character.isUpperCase(text.charAt(i))){
				result += (char)(((int)text.charAt(i) + shiftVal - 65)%26 + 65);
			}else{
				result += (char)(((int)text.charAt(i) + shiftVal - 97)%26 + 97);
			}
		}
		return result;
	}
}

class CaesarCipher{
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	public static void main(String[] args) throws IOException{
		// System.out.println();
		System.out.println("Enter the text to be encrypted");
		String text = br.readLine();
		System.out.println("Enter the shift value");
		int shiftVal = Integer.parseInt(br.readLine());
		Solution obj = new Solution();
		String cipherText = obj.caesarCipher(text , shiftVal);
		System.out.println("The entered ciphertext is : " + cipherText);
		System.out.println("The decrypted ciphertext is : " + obj.caesarCipher(cipherText , 26 - shiftVal));
	}
}

File: ICS/CaesarCipher.cpp
#include<bits/stdc++.h>
using namespace std;
#define mod 1000000007
#define rep(i,k,n) for(int i = k;i < n;++i)
#define pb push_back
#define mk make_pair
#define f first
#define s second

string encryptCaesarCipher(string text , int shiftVal){
	string result = "";
	for(int i = 0;i < text.length();i++){
		if(isupper(text[i])){
			result += char(int(text[i] + shiftVal - 65)%26 + 65);
		}else{
			result += char(int(text[i] + shiftVal - 97)%26 + 97);
		}
	}
	return result;
}

int main(){
	string text;
	int shiftVal;

	cout<<"Enter the text to be encrypted";
	cin>>text;
	cout<<"Enter the shiftValue";
	cin>>shiftVal;
	string cipherText = encryptCaesarCipher(text , shiftVal);
	cout<<"The ciphertext is : "<<cipherText<<endl;	
	cout<<"The decrypted ciphertext is : "<<encryptCaesarCipher(cipherText ,26-shiftVal);	
}